### PR TITLE
.github: Add snapshot generation on promotion

### PR DESCRIPTION
**Pull request by TeddyAndrieux**

### CODE REVIEW

File: .github/workflows/e2e-tests.yaml [hunk 1]
```diff
@@ -111,6 +121,27 @@ jobs:
...
+      - name: Get full MetalK8s version
+        if: inputs.generate-snapshots
+        run: |
+          source product.txt
+          echo "METALK8S_VERSION=$VERSION" >> $GITHUB_ENV
+      - name: Generate snapshot
+        if: inputs.generate-snapshots
+        uses: ./.github/actions/generate-snapshots
+        with:
+          snapshot-name: "metalk8s-${{ env.METALK8S_VERSION }}-rocky-8-${{ env.NAME }}-simple environment"
```
Comment (gdemonet) [inline .github/workflows/e2e-tests.yaml]: Do you remember why we introduced such complicated names? The suffix seems really redundant IMO (`simple environment` when we run `single-node`, and `3 nodes` when we run `multi-nodes`), wondering if we couldn't just give them a single "architecture" like in ARTESCA.

Comment (TeddyAndrieux) [reply .github/workflows/e2e-tests.yaml]: The idea was to have a lot more snapshots in the future, but right I will keep it simple now that we are (kind of) refactoring it

**Review by gdemonet — APPROVED** [review]
LGTM, I guess we could have factorized further and have a workflow handling a single spawn+install+tests+snapshot, using it twice (one for single-node, one for multi-nodes), but that's not critical I think

**Pull request merged**